superset-frontend/src/SqlLab/components/TableElement.tsx

```diff
@@ -17,31 +17,48 @@
  * under the License.
  */
 import React, { useState } from 'react';
-import PropTypes from 'prop-types';
 import Collapse from 'src/components/Collapse';
 import Card from 'src/components/Card';
 import ButtonGroup from 'src/components/ButtonGroup';
 import { t, styled } from '@superset-ui/core';
 import { debounce } from 'lodash';
 
 import { Tooltip } from 'src/components/Tooltip';
-import Icons from 'src/components/Icons';
 import CopyToClipboard from '../../components/CopyToClipboard';
 import { IconTooltip } from '../../components/IconTooltip';
-import ColumnElement from './ColumnElement';
+import ColumnElement, { ColumnKeyTypeType } from './ColumnElement';
 import ShowSQL from './ShowSQL';
 import ModalTrigger from '../../components/ModalTrigger';
 import Loading from '../../components/Loading';
 
-const propTypes = {
-  table: PropTypes.object,
-  actions: PropTypes.object,
-};
+interface Column {
+  name: string;
+  keys?: { type: ColumnKeyTypeType }[];
+  type: string;
+}
 
-const defaultProps = {
-  actions: {},
-  table: null,
-};
+interface Table {
+  name: string;
+  partitions?: {
+    partitionQuery: string;
+    latest: object[];
+  };
+  indexes?: object[];
+  selectStar?: string;
+  view?: string;
+  isMetadataLoading: boolean;
+  isExtraMetadataLoading: boolean;
+  columns: Column[];
+}
+
+interface TableElementProps {
+  table: Table;
+  actions: {
+    removeDataPreview: (table: Table) => void;
+    removeTable: (table: Table) => void;
+  };
+  key: string | number;
```

Comment:
This looks like it might just be a string? Are there places where it is used as a number? (though it is odd that it is a string, since a key should be a number. . .)